### PR TITLE
Use real data at navbar airdrop hook

### PR DESCRIPTION
**Pull request by oskarvu**
Closes FRO-653

### CODE REVIEW

File: packages/app/src/features/navbar/logic/useAirdropInfo.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { airdropInfo } from '@/domain/wallet/airdropInfo'
+
+import { AirdropInfo } from '../types'
+
+export function useAirdropInfo(): AirdropInfo {
+  const { address } = useAccount()
+
+  const result = useQuery({
+    ...airdropInfo(address && CheckedAddress(address)),
+  })
```
Comment (yivlad) [inline packages/app/src/features/navbar/logic/useAirdropInfo.ts:15]:
```suggestion
  const result = useQuery(
    airdropInfo(address && CheckedAddress(address)),
  )
```

File: packages/app/src/features/navbar/components/airdrop-badge/AirdropDetails.tsx [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
         <div className="flex items-center gap-2">
           <img src={assets.sparkIcon} className="h-7 lg:h-6" />
-          <div className="text-basics-black text-base font-semibold">
-            {USD_MOCK_TOKEN.format(amount, { style: 'auto' })} SPK
-          </div>
+          {isLoading ? (
+            <Skeleton className="h-5 w-7" />
+          ) : (
+            <div className="text-basics-black text-base font-semibold">
+              {USD_MOCK_TOKEN.format(amount, { style: 'auto' })} SPK
```
Comment (yivlad) [inline packages/app/src/features/navbar/components/airdrop-badge/AirdropDetails.tsx:21]: I'd go with local simple compact formatter here. It doesn't need to have precision of token formatting

**Review by yivlad — APPROVED** [review]

File: packages/app/src/domain/wallet/airdropInfo.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { queryOptions, skipToken } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import { CheckedAddress } from '../types/CheckedAddress'
+
+const airdropInfoResponseSchema = z.object({
+  token_reward: z.string(),
```
Comment (krzkaczor) [inline packages/app/src/domain/wallet/airdropInfo.ts:7]: wouldn't it be better to already use normalizedNumber validator here? I am pretty sure that we already created is somewhere.

File: packages/app/src/features/navbar/logic/useAirdropInfo.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { airdropInfo } from '@/domain/wallet/airdropInfo'
+
+import { AirdropInfo } from '../types'
+
+export function useAirdropInfo(): AirdropInfo {
+  const { address } = useAccount()
+
+  const result = useQuery(airdropInfo(address && CheckedAddress(address)))
+
+  const amount = NormalizedUnitNumber(result.data?.token_reward ?? 0)
```
Comment (krzkaczor) [inline packages/app/src/features/navbar/logic/useAirdropInfo.ts:15]: this should be simplified after airdropInfoQuery is improved.

File: packages/app/src/features/navbar/components/airdrop-badge/AirdropDetails.tsx (hunk 2 above)
Comment (krzkaczor) [reply packages/app/src/features/navbar/components/airdrop-badge/AirdropDetails.tsx:21]: How about we create `SPK_MOCK_TOKEN`?

```
export const SPK_MOCK_TOKEN = new Token({
  address: CheckedAddress(zeroAddress),
  symbol: TokenSymbol('SPK'),
  name: 'Spark Token',
  decimals: 18,
  unitPriceUsd: '0',
})
```

File: packages/app/src/features/navbar/utils/formatCompact.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { NormalizedUnitNumber } from '@/domain/types/NumericValues'
+
+export function formatCompact(value: NormalizedUnitNumber): string {
```
Comment (krzkaczor) [inline packages/app/src/features/navbar/utils/formatCompact.ts:3]: could use SPK_MOCK_TOKEN compact formatting.

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**